howso/input_validation.amlg

```diff
@@ -104,4 +104,197 @@
 		errors
 	)
 
+	;Validates parameter inputs for features that are in datetime format
+	;verifies that the datetime format matches what is set in feature attributes
+	; this method checks the following parameters:
+	; - initial_values
+	; - series_stop_map(s)
+	; - init_time_steps
+	; - final_time_steps
+	;
+	;parameters:
+	; single_series: indicates if the method is called for a single series (from #react_series) or for multiple series (from #batch_react_series)
+	#!ValidateDateTimeInputs
+	(let
+		(assoc
+			single_series (false)
+
+			;not parameters
+			epoch_value (null)
+			datetime_value (null)
+			invalid_value_found (false)
+		)
+
+		(if initial_features
+			(if single_series
+				;not batch_react_series, initial_values is a 1-d list
+				(map
+					(lambda
+						(if (contains_index !featureDateTimeMap (current_index))
+							(call !CheckDateTimeFormat (assoc
+								datetime_feature (current_index 1)
+								datetime_string (current_value 1)
+								parameter_name "initial_values"
+							))
+						)
+					)
+					(zip initial_features initial_values)
+				)
+
+				;else initial_values is a list of lists
+				(map
+					(lambda
+						(map
+							(lambda
+								(if (contains_index !featureDateTimeMap (current_index))
+									(call !CheckDateTimeFormat (assoc
+										datetime_feature (current_index 1)
+										datetime_string (current_value 1)
+									))
+								)
+							)
+							;zip features to each list of values
+							(zip initial_features (current_value))
+						)
+					)
+					initial_values
+				)
+			)
+		)
+		(if invalid_value_found
+			(conclude (conclude (conclude
+				(call !Return (assoc
+					errors (list (concat "Incorrect datetime format used in initial_values.") )
+				))
+			)))
+		)
+
+		(if init_time_steps
+			(call !CheckDateTimeFormat (assoc
+				datetime_feature !tsTimeFeature
+				datetime_string (first init_time_steps)
+			))
+		)
+		(if invalid_value_found
+			(conclude (conclude (conclude
+				(call !Return (assoc
+					errors (list (concat "Incorrect datetime format used in init_time_steps.") )
+				))
+			)))
+		)
+
+		(if final_time_steps
+			(call !CheckDateTimeFormat (assoc
+				datetime_feature !tsTimeFeature
+				datetime_string (first final_time_steps)
+			))
+		)
+		(if invalid_value_found
+			(conclude (conclude (conclude
+				(call !Return (assoc
+					errors (list (concat "Incorrect datetime format used in final_time_steps.") )
+				))
+			)))
+		)
+
+		(if series_stop_map
```

Comment:
get rid of this block of code, lines 200-225 altogether, combine with the block below